### PR TITLE
Included Private ai stack basic sanity test

### PR DESCRIPTION
**Pull request by yarunachalam**
- Related ticket: https://jira.suse.com/browse/SUSEAI-8
- Verification run: waiting for job template PR to be merged. So, I can do verification run

### CODE REVIEW

File: tests/publiccloud/aistack_basic.pm [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        timeout => 100 );
+    $instance->ssh_assert_script_run( 'sudo systemctl start rke2-server',
+        timeout => 100 );
+    $instance->ssh_assert_script_run( 'sudo systemctl status rke2-server',
+        timeout => 100 );
+    record_info('Installed RKE2');
+
+    #Install kubectl
+    $instance->ssh_assert_script_run(
+        'curl -LO "https://dl.k8s.io/release/v1.26.0/bin/linux/amd64/kubectl',
```
Comment (rbmarliere) [inline tests/publiccloud/aistack_basic.pm:58]: Should the version be hard-coded this way?

Comment (yarunachalam) [reply tests/publiccloud/aistack_basic.pm:58]: This is draft for testing. Planning to do as a parameter.

**Comment by m-dati** [comment]
From what was used in pub.cl. and containers tests, it could be that this code flow could be rewritten in other way. 
I mean that pre-installation of containers in publiccloud tests already are available in our tests after prepare_instance , registratio, ssh_interactive_start, etc... See i.e. OSD:Pub.Clo. Latest->publiccloud_container tests.

Probably it is useful some study on those test modules structures, before of this unless a new project is going on that I am not aware :) 

@asmorodskyi comments would be useful here.

**Comment by m-dati** [comment]
Please, see here below a summary of my suggestions and notes:

Taking as basic similitude the code in `tests/publiccloud/run_ltp.pm`:

1) on top of module I see probably missing the @_ input parameter initialization:
`my ($self, $args) = @_` (eventually then remember `$self->item ...`);

2) in place of select_serial_terminal(), I'd suggest the use of its wrapper publiccloud::ssh_interactive -> `select_host_console()`, including more checks.

3) like runltp L99, suggested the "if-branch instance assignment from input, otherwise creation when not available" ($qam empty by default will run creation). 

4) using n.3, then run registercloudguest($instance) with checks like in run_ltp.

5) be aware if the system would be transactional you need to add dedicated install (tr-up) and reboot actions, being Private-AI defined on `SL micro`.

File: tests/publiccloud/aistack_basic.pm (hunk 1 above)
Comment (yarunachalam) [reply tests/publiccloud/aistack_basic.pm]: done

File: tests/publiccloud/aistack_basic.pm [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    my $aws_test = get_var('PUBLIC_CLOUD_AWS_TEST', 'basic_sanity');
+    if ($aws_test) {
+        $instance = $self->{my_instance} = $args->{my_instance};
+        $provider = $self->{provider} = $args->{my_provider};    # required for cleanup
+    } else {
+        $provider = $self->{provider} = $self->provider_factory();    # required for cleanup
+        $instance = $self->{my_instance} = $provider->create_instance(check_guestregister => is_openstack ? 0 : 1);
+    }
+
+    registercloudguest($instance) if is_byos();
```
Comment (m-dati) [inline tests/publiccloud/aistack_basic.pm:35]: Likewise in same place in run_ltp, by experienced old fault, we need to add the same second condition too (see !$qam) :slightly_smiling_face:.

File: tests/publiccloud/aistack_basic.pm [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    my $ins_helm = get_var('HELM_URL');
+    #my $add_repo = get_var('HELM_REPO');
+   
+
+    #Create aws instance
+    my $provider;
+    my $instance;
+    select_host_console();
+
+    my $aws_test = get_var('PUBLIC_CLOUD_AWS_TEST', 'basic_sanity');
```
Comment (m-dati) [inline tests/publiccloud/aistack_basic.pm:26]: Being (Private) AI the effective context needing such test, IMHO related parameters should also evoke and reference to AI, having for publiccloud specific parameter with similar meanings.
This is why I suggest here and elsewhere needed in that project, a different name than `...AWS_TEST`, where even AWS could in future be another CSP, to be more related to the AI project.
It should mean: _p.c.instance available/1 or not/0_, like i.e. longer but clear `PUBLIC_CLOUD_INSTANCE_FOR_AI` or else.

File: tests/publiccloud/aistack_basic.pm [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    #Update and install dependencies
+    $instance->ssh_assert_script_run('sudo zypper ref; sudo zypper -n up',
+        timeout => 1000);
+    record_info('Refresh and update');
+    sleep 90;    # wait for a bit for zypper to be available
+
+    #Install Docker (required for RKE2)
+
+    $instance->ssh_assert_script_run('sudo transactional-update -n pkg install -y curl',
+        timeout => 100);
```
Comment (m-dati) [inline tests/publiccloud/aistack_basic.pm:46]: Please remember that tr-up changes need for reboot to be applied. Anyway being an heavy action, better to check if package eventually already present, you can skip it.
Anyway, we already have `lib/transactional.pm::trup_install()` doing that work.

File: tests/publiccloud/aistack_basic.pm [hunk 5]
```diff
@@ -0,0 +1,89 @@
+use Mojo::Base 'publiccloud::basetest';
```
Comment (m-dati) [inline tests/publiccloud/aistack_basic.pm:1]: CI still claiming tidy not ok.
Before to push your changes I suggest to run in your local repo the command `tools/tidy <changed_file>`

File: tests/publiccloud/aistack_basic.pm (hunk 2 above)
Comment (yarunachalam) [reply tests/publiccloud/aistack_basic.pm:35]: Fixed

File: tests/publiccloud/aistack_basic.pm (hunk 3 above)
Comment (yarunachalam) [reply tests/publiccloud/aistack_basic.pm:26]: Fixed

File: tests/publiccloud/aistack_basic.pm (hunk 4 above)
Comment (yarunachalam) [reply tests/publiccloud/aistack_basic.pm:46]: Should I replace it like this
`#Install Docker (required for RKE2)

    record_info 'Install tar', 'Install package curl using transactional server and reboot';
    trup_install "curl";
    record_info 'Install tar', 'Install package docker using transactional server and reboot';
    trup_install "docker";
    $instance->ssh_assert_script_run('sudo systemctl start docker',
        timeout => 100);
    $instance->ssh_assert_script_run('sudo systemctl enable docker',
        timeout => 100);
    $instance->ssh_assert_script_run('sudo systemctl status docker',
        timeout => 100);
    record_info('Installed Docker');
`

Comment (yarunachalam) [reply tests/publiccloud/aistack_basic.pm]: Fixed

**Comment by m-dati** [comment]
The CI detected tidy missing, see i.e. L#16 in function call's space before arguments. Needed: `tools/tidy tests/publiccloud/aistack_basic.pm`, rebase, push -f ...

File: tests/publiccloud/aistack_basic.pm (hunk 5 above)
Comment (yarunachalam) [reply tests/publiccloud/aistack_basic.pm:1]: Fixed

**Comment by yarunachalam** [comment]
trigger job https://gitlab.suse.de/qac/gravity-cannon/-/merge_requests/81 in progress

**Comment by m-dati** [comment]
Sorry, still a comment about CI failure :smile:
1. https://github.com/os-autoinst/os-autoinst-distri-opensuse/actions/runs/11027316476/job/30625398539?pr=2_0015#step:6:15 ,
about migration_autoyast.yaml and mau-extratests2.yaml errors;

2. https://github.com/os-autoinst/os-autoinst-distri-opensuse/actions/runs/11027316416/job/30625398972?pr=20015#step:5:17 , 
see the [OOO test](https://openqa.opensuse.org/tests/4507002), see in _[previous results](https://openqa.opensuse.org/tests/4504554#next_previous)_ last passed is before your MR run, all other next failed: somethings in schedule.

File: tests/publiccloud/aistack_basic.pm [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+    }
+    else {
+        $provider = $self->{provider} =
+          $self->provider_factory();    # Required for cleanup
+        $instance = $self->{my_instance} = $provider->create_instance(
+            check_guestregister => is_openstack ? 0 : 1);
+    }
+
+    registercloudguest($instance) if is_byos();
+    registercloudguest($instance) if (is_byos() && !$aws_ai_test);
```
Comment (m-dati) [inline tests/publiccloud/aistack_basic.pm:101]: It seems you didn't deleted previous line100 old-version :slightly_smiling_face: : typo?

File: tests/publiccloud/aistack_basic.pm [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    # Validate OpenWebUI
+    #validate_openwebui($admin_username, $admin_password, $openwebui_hostname);
+    # Verify Ollama api endpoint
+    #verify_api_endpoint();
+    # Milvus DB Test
+    #milvus_db_test();
+    # Milvus DB Test
+    #milvus_db_test();
+    # OpenWebUI Integration test
+    #test_openwebui_interaction();
```
Comment (m-dati) [inline tests/publiccloud/aistack_basic.pm:115]: same commen as above: can we delete deactivated code?

File: tests/publiccloud/aistack_basic.pm [hunk 8]
```diff
@@ -0,0 +1,132 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Basic aistack test
+#
+#   This test does the following
+#    - Create VM in EC2 SLE-Micro-6-0-BYOS.x86_64-1.0.0-EC2-Build1.36.raw.xz 
+#    - Installs required dependency to install aistack helmchart and containers
```
Comment (m-dati) [inline tests/publiccloud/aistack_basic.pm:10]: One spaces here at each line end, to delete for C.I. tidy :)

**Comment by yarunachalam** [comment]
Closing this ticket and redirected the aistack test in new ticket https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20427

**Pull request closed without merging**